taiga2/controllers/endpoint.py

```diff
@@ -532,12 +531,14 @@ def create_upload_session_file(uploadMetadata, sid):
         if data_file is None:
             api_error("Unknown taiga ID: " + existing_taiga_id)
 
-        # When adding a reference to an existing Taiga datafile, it is possible
-        # to provide a new set of custom metadata in the form of key/value pairs.
-        # If new custom_metadata is not provided, copy the key/value pairs from the
-        # original file.
         if custom_metadata == None:
             custom_metadata = data_file.custom_metadata
```

Comment:
This didn't work, because data_file always resolved to the original underlying s3 file. If file_A.1 was s3 with custom_metadata == None, and file_A.2 was virtual with custom_metadata: { "test_key": "test_val"}, adding file_A.3 as virtual would result in stomping out the custom_metadata field that had been added to file_A.2.